profiles/permissions_test.py

```diff
@@ -47,6 +59,149 @@ def test_cant_edit_if_not_owner(self):
         other_user = UserFactory.create()
         for method in ('POST', 'PATCH', 'PUT'):
             with mute_signals(post_save):
-                profile = ProfileFactory.create()
+                profile = ProfileFactory.create(account_privacy=Profile.PUBLIC_TO_MM)
+
             request = Mock(method=method, user=other_user)
             assert not perm.has_object_permission(request, None, profile)
+
+
+# pylint: disable=no-self-use
+class CanSeeIfNotPrivateTests(TestCase):
+    """
+    Tests for CanSeeIfNotPrivate permissions
+    """
+
+    def setUp(self):
+        super(CanSeeIfNotPrivateTests, self).setUp()
+        with mute_signals(post_save):
+            self.other_user = other_user = UserFactory.create(username="test")
+            username = "{}_edx".format(other_user.username)
+            other_user.social_auth.create(
+                provider=EdxOrgOAuth2.name,
+                uid=username
+            )
+            self.social_auth = self.other_user.social_auth
+            self.other_user_profile = ProfileFactory.create(user=other_user, verified_micromaster_user=False)
+
+    def test_cant_view_if_privacy_is_private(self):
+        """
+        Users are not supposed to view private profiles.
+        """
+        perm = CanSeeIfNotPrivate()
+
+        with mute_signals(post_save):
+            profile_private = ProfileFactory.create(account_privacy=Profile.PRIVATE)
+
+        request = Mock(user=self.other_user)
+        result = Mock(return_value=profile_private)
+
+        with patch('rest_framework.generics._get_object_or_404', result):
+            view = MagicMock(name='kwargs', return_value={'user': self.social_auth})
+            with self.assertRaises(Http404):
+                perm.has_permission(request, view)
+
+    def test_cant_view_if_anonymous_user(self):
+        """
+        Anonymous are not supposed to view public_to_mm profiles.
+        """
+        perm = CanSeeIfNotPrivate()
+        with mute_signals(post_save):
+            profile_private = ProfileFactory.create(account_privacy=Profile.PUBLIC_TO_MM)
+
+        request = Mock(user=Mock(return_value=Mock(method='is_anonymous', return_value=True)))
+        result = Mock(return_value=profile_private)
+
+        with patch('rest_framework.generics._get_object_or_404', result):
```

Comment:
i am mocking this because one user is viewing profile of the other user and here we are testing permission not view it self. So we dont have reference of view